skxray/tests/test_diff_roi_choice.py

```diff
@@ -0,0 +1,332 @@
+                             4, 4, 5, 5, 6, 6, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+                             4, 5, 5, 5, 5, 6, 6, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+                             5, 5, 5, 6, 6, 6, 5, 5, 5, 5, 6, 6, 6, 5, 5, 5,
+                             5, 5, 5, 5, 5, 5, 5, 5, 6, 6, 6, 6, 5, 5, 5, 5,
+                             5, 5, 5, 5, 5, 6, 6, 6])
+
+    numstep_pixels_m = np.array([34, 35, 45, 57, 62, 47])
+
+    qstep_ring_val_m = np.array([[2.5, 4.5],
+                                 [5.5, 7.5],
+                                 [8.5, 10.5],
+                                 [11.5, 13.5],
+                                 [14.5, 16.5],
+                                 [17.5, 19.5]])
+
+    pixelstep_list_m = ([0, 1, 2, 3, 4, 5, 6, 9, 10, 12, 13, 15, 16, 18,  19,
+                         21, 22, 25, 31, 32, 34, 35, 37, 38, 40, 41, 43, 44,
+                         46, 47, 50, 56, 57, 59, 60, 62, 63, 65, 66, 68, 69,
+                         71, 72, 75, 81, 82, 84, 85, 87, 88, 90, 91, 93, 94,
+                         96, 97, 100, 106, 107, 109, 110, 112, 113, 115, 116,
+                         118, 119, 121, 122, 125, 131, 132, 134, 135, 137, 138,
+                         140, 141, 143, 144, 146, 147, 150, 151, 152, 153, 154,
+                         155, 156, 159, 160, 162, 163, 165, 166, 168, 169, 171,
+                         172, 176, 177, 178, 179, 180, 183, 184, 187, 188, 190,
+                         191, 193, 194, 196, 197, 207, 208, 209, 211, 212, 214,
+                         215, 216, 218, 219, 221, 222, 225, 226, 227, 228, 229,
+                         230, 231, 232, 233, 235, 236, 237, 239, 240, 242, 243,
+                         245, 246, 247, 250, 251, 252, 253, 254, 255, 256, 259,
+                         260, 261, 263, 264, 265, 267, 268, 270, 271, 283, 284,
+                         285, 287, 288, 289, 291, 292, 295, 296, 300, 301, 302,
+                         303, 304, 305, 306, 307, 308, 309, 311, 312, 313, 315,
+                         316, 317, 319, 320, 325, 326, 327, 328, 329, 330, 331,
+                         332, 335, 336, 337, 340, 341, 343, 344, 345, 358, 359,
+                         360, 361, 364, 365, 368, 369, 375, 376, 377, 378, 379,
+                         380, 381, 382, 383, 384, 385, 387, 388, 389, 390, 392,
+                         393, 400, 401, 402, 403, 404, 405, 406, 407, 408, 411,
+                         412, 413, 416, 417, 418, 434, 435, 436, 437, 440, 441,
+                         442, 450, 451, 452, 453, 454, 455, 456, 457, 458, 459,
+                         460, 463, 464, 465, 466, 475, 476, 477, 478, 479, 480,
+                         481, 482, 483, 487, 488, 489]),
+
+    assert_almost_equal(qstep_ring_val, qstep_ring_val_m)
+    assert_array_equal(numstep_pixels, numstep_pixels_m)
+    assert_array_equal(qstep_inds, np.ravel(qstep_inds_m))
+    assert_array_equal(pixelstep_list, pixelstep_list_m)
+
+
+def test_roi_rings_diff_steps():
```

Comment:
After fixing the bug

fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12, 6), tight_layout=True)
 test_demo(ax1, calibrated_center, qd_inds, qd_ring_val, pixeld_list, img_dim)
 #test_demo(ax2, calibrated_center, qd_inds, qd_ring_val, pixeld_list, img_dim)
 plt.show()

![qd_inds](https://cloud.githubusercontent.com/assets/5701652/6142725/f5f27026-b189-11e4-8fd0-18282b6e9ab9.png)